Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/RetrieveProjectController.java

```diff
@@ -106,16 +107,38 @@ public ModelAndView getProjectList(Principal principal) throws QuadrigaStorageEx
                 }
             }
         }
-
-        // Fetch all the projects for which the user is associated workspace
-        // collaborator
+    
+        // Fetch all the projects for which the user is associated workspace collaborator.
         List<IProject> projectListAsWSCollaborator = projectManager.getProjectListAsWorkspaceCollaborator(userName);
+        IProject tempProject = null;;
+        List<IProjectWorkspace> tempProjectWorkspaces = new ArrayList<IProjectWorkspace>();
+        List<IWorkspaceCollaborator> tempWorkspaceCollaborators;
         if (projectListAsWSCollaborator != null) {
             for (IProject p : projectListAsWSCollaborator) {
+               // Process the project details if it has not been evaluated.
                 if (!projectIds.contains(p.getProjectId())) {
-                    fullProjects.add(projectManager.getProjectDetails(p.getProjectId()));
+                    // Get details of the project using the project id.
+                    tempProject = projectManager.getProjectDetails(p.getProjectId());
```

Comment:
why do you retrieve the project details again? You already have a project object.